nearcore/src/lib.rs

```diff
@@ -71,84 +71,17 @@ pub fn get_default_home() -> PathBuf {
     PathBuf::default()
 }
 
-/// Returns the path of the DB checkpoint.
-/// Default location is the same as the database location: `path`.
-fn db_checkpoint_path(path: &Path, near_config: &NearConfig) -> PathBuf {
-    let root_path =
-        if let Some(db_migration_snapshot_path) = &near_config.config.db_migration_snapshot_path {
-            assert!(
-                db_migration_snapshot_path.is_absolute(),
-                "'db_migration_snapshot_path' must be an absolute path to an existing directory."
-            );
-            db_migration_snapshot_path.clone()
-        } else {
-            path.to_path_buf()
-        };
-    root_path.join(DB_CHECKPOINT_NAME)
-}
-
-const DB_CHECKPOINT_NAME: &str = "db_migration_snapshot";
-
-/// Creates a consistent DB checkpoint and returns its path.
-/// By default it creates checkpoints in the DB directory, but can be overridden by the config.
-fn create_db_checkpoint(path: &Path, near_config: &NearConfig) -> Result<PathBuf, anyhow::Error> {
-    let checkpoint_path = db_checkpoint_path(path, near_config);
-    if checkpoint_path.exists() {
-        return Err(anyhow::anyhow!(
-            "Detected an existing database migration snapshot: '{}'.\n\
-             Probably a database migration got interrupted and your database is corrupted.\n\
-             Please replace the contents of '{}' with data from that checkpoint, delete the checkpoint and try again.",
-            checkpoint_path.display(),
-            path.display()));
-    }
-
-    let db = RocksDB::new(path)?;
-    let checkpoint = db.checkpoint()?;
-    info!(target: "near", "Creating a database migration snapshot in '{}'", checkpoint_path.display());
-    checkpoint.create_checkpoint(&checkpoint_path)?;
-    info!(target: "near", "Created a database migration snapshot in '{}'", checkpoint_path.display());
-
-    Ok(checkpoint_path)
-}
-
 /// Function checks current version of the database and applies migrations to the database.
 pub fn apply_store_migrations(path: &Path, near_config: &NearConfig) {
     let db_version = get_store_version(path);
     if db_version > near_primitives::version::DB_VERSION {
         error!(target: "near", "DB version {} is created by a newer version of neard, please update neard or delete data", db_version);
         std::process::exit(1);
     }
-
     if db_version == near_primitives::version::DB_VERSION {
         return;
     }
 
-    // Before starting a DB migration, create a consistent snapshot of the database. If a migration
```

Comment:
hmm why do we want to delete this? Even with the graceful shutdown here, receiving SIGINT in the middle of a migration would still leave the DB in a bad state, right?